File: Backend/local/src/main/java/com/locales/service/LocalService.java
package com.locales.service;

import java.util.Map;

import org.springframework.http.ResponseEntity;

import com.locales.model.Local;


public interface LocalService {
	public ResponseEntity<Map<String, Object>> listarLocales();
	public ResponseEntity<Map<String, Object>> listarLocalesPorId(Long id_local);
	public ResponseEntity<Map<String, Object>> agregarLocal(Local local);
	public ResponseEntity<Map<String, Object>> editarLocal(Local local, Long id_local);
	public ResponseEntity<Map<String, Object>> eliminarLocal(Long id_local);
	
	public ResponseEntity<Map<String, Object>> eliminarLocalEnable(Long id_local);
	public ResponseEntity<Map<String, Object>> listarLocalesEnable();

}

File: Frontend/src/app/servicio/local.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { HttpClient } from '@angular/common/http';
import { Local } from '../modelos/local';

@Injectable({
  providedIn: 'root'
})
export class LocalService {
  private baseUrl = environment.apiUrl;
  private url: string = `${this.baseUrl}/locales`

  constructor(private http:HttpClient) { }

  listarLocales(): Observable<any>{
    return this.http.get<any>(this.url);
  }

  listarLocalesEnable(): Observable<any> {
    return this.http.get<any>(`${this.url}/enable`);
  }

  listarLocalesPorId(id_local: number): Observable<Local> {
    return this.http.get<Local>(`${this.url}/${id_local}`);
  }  

  agregarLocal(local: Local): Observable<any> {
    return this.http.post<any>(this.url, local);
  }

  editarLocal(id_local: number, local: Local): Observable<any> {
    return this.http.put<any>(`${this.url}/${id_local}`, local);
  }

  eliminarLocalEnable(local: Local): Observable<any> {
    return this.http.put<Local>(`${this.url}/enable/${local.id_local}`, local);
  }

}
